File: frontend/lib/models/create_models/create_clothing_item_model.dart
import 'package:flameguardlaundry/models/gear_condition.dart';
import 'package:flutter/widgets.dart';
import 'package:json_annotation/json_annotation.dart';

part 'create_clothing_item_model.g.dart';

@JsonSerializable()
@immutable
class CreateClothingItemModel {
  final String variantId;
  final String? identifier;
  final String? storageLocationId;
  final GearCondition condition;
  final DateTime purchaseDate;
  final DateTime? retirementDate;

  const CreateClothingItemModel({
    required this.variantId,
    this.identifier,
    this.storageLocationId,
    required this.condition,
    required this.purchaseDate,
    required this.retirementDate,
  });

  factory CreateClothingItemModel.fromJson(Map<String, dynamic> json) =>
      _$CreateClothingItemModelFromJson(json);
  Map<String, dynamic> toJson() => _$CreateClothingItemModelToJson(this);
}

File: backend/Shared/Models/CreateClothingItemModel.cs
using FireInvent.Contract;
using System.ComponentModel.DataAnnotations;

namespace FireInvent.Shared.Models;

public record CreateClothingItemModel
{
    [Required]
    public Guid VariantId { get; init; }

    [MaxLength(ModelConstants.MaxStringLength)]
    public string? Identifier { get; init; } = string.Empty;

    public Guid? StorageLocationId { get; init; }

    [Required]
    public GearCondition Condition { get; init; }

    [Required]
    public DateTime PurchaseDate { get; init; }

    public DateTime? RetirementDate { get; init; }
}
